src/NuGet.Services.AzureSearch/DocumentUtilities.cs

```diff
@@ -116,13 +117,19 @@ public static string GetSearchFilterString(SearchFilters searchFilters)
             {
                 document.LicenseUrl = package.LicenseUrl;
             }
+
+            // TODO: detect whether the package entity has an icon file once there is schema for it.
+            // https://github.com/NuGet/NuGetGallery/issues/7261
+            // https://github.com/nuget/nugetgallery/issues/7064
         }
 
         public static void PopulateMetadata(
             IBaseMetadataDocument document,
             string normalizedVersion,
             PackageDetailsCatalogLeaf leaf,
-            Uri galleryBaseUrl)
+            Uri galleryBaseUrl,
```

Comment:
Consider making this class non-static and passing these configs through constructor. You can make `FlatContainerPackagePathProvider` a field that's created by dependency injection too.

Reply:
Done. Left `FlatContainerPackagePathProvider` as a non-field to respect the option's reload semantics.